zebra-chain/src/serialization/error.rs

```diff
@@ -50,4 +50,8 @@ pub enum SerializationError {
     /// rule](https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus).
     #[error("transaction balance is non-zero but doesn't have Sapling shielded spends or outputs")]
     BadTransactionBalance,
+
+    ///  Invalid BLOCK_MAINNET_******_BYTES vector ID given to vector fetcher
+    #[error("invalid mainnet / testnet bytes id given to vector fetcher")]
+    UnsupportedVersion(u32),
```

Comment:
Could we remove this test error, call `ok()` to convert the `Result` from `zcash_deserialize_into()` to an Option where needed, and return `None` when the arguments don't match any test blocks for the network?

Alternatively, could we put this variant behind `#[cfg(test)]`?

I'll add a comment below with a suggestion.

Reply:
removed error type completely as per suggestion below.